src/Emulator/Cores/RiscV/OpenTitan_PlatformLevelInterruptController.cs

```diff
@@ -0,0 +1,141 @@
+                            // TODO: Build register value from pending sources IRQs
+                            return 0;
+                        }));
+            }
+
+            long InterruptSourceMode0 = (long)Registers.InterruptPending0 + (InterruptPendingRegisterCount * 4);
+            for(var i = 0; i < InterruptSourceModeCount; i++)
+            {
+                long InterruptSourceModeOffset = InterruptSourceMode0 + i *4;
+                this.Log(LogLevel.Noisy, "Creating InterruptSourceMode{0}[0x{1:X}]",i, InterruptSourceModeOffset);
+                registersMap.Add(InterruptSourceModeOffset, new DoubleWordRegister(this)
+                    .WithValueField(0, 32, FieldMode.Read | FieldMode.Write, valueProviderCallback: (_) => {
+                            return 0;
+                        }, writeCallback: (_, value) => {
+                            this.Log(LogLevel.Noisy, $"Write InterruptSourceMode0 0x{value:X}");
+                        }));
+            }
+
+            long Source0Priority = InterruptSourceMode0 + (InterruptSourceModeCount * 4);
+            for(var i = 0; i < numberOfSources; i++)
+            {
+                var j = i;
+                long SourcePriorityOffset = Source0Priority +  (4 * i);
+                this.Log(LogLevel.Noisy, "Creating SourcePriority{0}[0x{1:X}]", i, SourcePriorityOffset);
+                registersMap[SourcePriorityOffset] = new DoubleWordRegister(this)
+                    .WithValueField(0, 3,
+                                    valueProviderCallback: (_) => irqSources[j].Priority,
+                                    writeCallback: (_, value) =>
+                                    {
+                                        irqSources[j].Priority = value;
+                                        RefreshInterrupts();
+                                    });
+            }
+
+            // Algorithm for offset is pulled from the HJSON file within OpenTitan rv_plic
+            long Target0MachineEnablesOffset = (long)(0x100*(Math.Ceiling(((numberOfSources)*4+8*Math.Ceiling((numberOfSources + 1)/32.0))/0x100)) + 0*0x100);
+            int TargetMachineEnableCount = (int)Math.Ceiling((numberOfSources) / 32.0);
+            int maximumSourceDoubleWords = (int)Math.Ceiling((numberOfSources) / 32.0) * 4;
+            this.Log(LogLevel.Noisy, $"TargetMachineEnableCount 0x{TargetMachineEnableCount:X}");
+            this.Log(LogLevel.Noisy, $"maximumSourceDoubleWords 0x{maximumSourceDoubleWords:X}");
+            this.Log(LogLevel.Noisy, $"Target0MachineEnablesOffset 0x{Target0MachineEnablesOffset:X}");
+
+            for(var i = 0; i < numberOfTargets; i++)
+            {
+                long TargetMachineEnablesOffset = Target0MachineEnablesOffset + i * maximumSourceDoubleWords + (i * 4 * 3);
+                AddTargetEnablesRegister(registersMap, TargetMachineEnablesOffset, (uint)i, (PrivilegeLevel)0, numberOfSources - 1);
```

Comment:
Note I added the -1 here. This way the number of sources in the platform config file matches with how opentitan generates the PLIC.

Example:
opentitan/hw/ip/rv_plic/util/reg_rv_plic.py --sources 96 --targets 1 opentitan/hw/ip/rv_plic/data/rv_plic.hjson.tpl  > rv_plic.hjson &&  opentitan/util/regtool.py -D rv_plic.hjson | less